test/code/providers/Container_ImageInventory_Class_Provider_UnitTest.cpp

```diff
@@ -162,8 +206,8 @@ class ImageInventoryTest : public CppUnit::TestFixture
 		for (unsigned i = 0; i < images.size(); i++)
 		{
 			bool flag = false;
-			int rc = mbstowcs(currentId, cJSON_GetObjectItem(images[i], "InstanceID")->valuestring, (sizeof(currentId)/sizeof(currentId[0])) - 1);
-			if (rc > 0 && rc < sizeof(currentId)/sizeof(currentId[0])) currentId[rc] = 0;
+			int rc = (int)mbstowcs(currentId, cJSON_GetObjectItem(images[i], "InstanceID")->valuestring, (sizeof(currentId)/sizeof(currentId[0])) - 1);
```

Comment:
It returns an unsigned value, but uses a negative value to indicate an error. I cast to int so that the rc > 0 is meaningful.